src/message/__tests__/messageActionSheet-test.js

```diff
@@ -58,6 +58,34 @@ describe('constructActionButtons', () => {
 });
 
 describe('constructTopicActionButtons', () => {
+  test('show mark as read if topic is unread', () => {
+    const stream = eg.makeStream({ name: 'test stream', streamId: 123 });
+    const subscriptions = [{ ...eg.subscription, ...stream }];
+    const unreadStreams = eg.makeUnreadStreamsState([
+      { streamId: 123, topics: [{ topicName: 'test topic', messageIds: [1, 2] }] },
+    ]);
+    const buttons = constructTopicActionButtons({
+      backgroundData: { ...baseBackgroundData, subscriptions, unreadStreams },
+      stream: 'test stream',
+      topic: 'test topic',
+    });
+    expect(buttonTitles(buttons)).toContain('Mark topic as read');
+  });
+
+  test('do not show mark as read if topic is read', () => {
+    const stream = eg.makeStream({ name: 'test stream', streamId: 123 });
+    const subscriptions = [{ ...eg.subscription, ...stream }];
+    const unreadStreams = eg.makeUnreadStreamsState([
+      { streamId: 123, topics: [{ topicName: 'unread topic', messageIds: [1, 2] }] },
```

Comment:
I don't think there's a reason to include this stream in `unreadStreams`, or even to override `unreadStreams` at all, since this test doesn't look at it.

Reply:
I kept it here to make sure that having any random topic in `unreadStream` doesn't affect the `markTopicAsRead` button to the actionsheet for another topic (this isn't true in the current scenario but may be true for a possible future bad implementation.)
I haven't removed it in the new revision, give me another cue if you think that this is unnecessary and I will remove it.